File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmBugs.vb
Partial Public Class frmBugs
    Inherits Form

    Private oBugService As BugService = New BugService()
    Private criticidadService As CriticidadService = New CriticidadService()
    Private prioridadService As PrioridadService = New PrioridadService()
    Private productoService As ProductoService = New ProductoService()
    Private usuarioService As UsuarioService = New UsuarioService()
    Private estadoService As EstadoService = New EstadoService()
    Friend row_selected As DataGridViewRow

    Public Sub New()
        InitializeComponent()
        dgvBugs.AutoGenerateColumns = False
    End Sub

    Private Sub frmBugs_Load(ByVal sender As Object, ByVal e As EventArgs)
        GUIHelper.getHelper().llenarCombo(cboEstados, estadoService.consultarEstados(), "nombre", "id_estado")
        GUIHelper.getHelper().llenarCombo(cboPrioridades, prioridadService.consultarPrioridades(), "nombre", "id_prioridad")
        GUIHelper.getHelper().llenarCombo(cboCriticidades, criticidadService.consultarCriticidades(), "nombre", "id_criticidad")
        GUIHelper.getHelper().llenarCombo(cboAsignadoA, usuarioService.consultarUsuarios(), "nombre", "id_usuario")
        GUIHelper.getHelper().llenarCombo(cboProductos, productoService.consultarProductos(), "nombre", "id_producto")
    End Sub

    Private Sub btnConsultar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConsultar.Click
        Dim parametros As List(Of Object) = New List(Of Object)()
        Dim fechaDesde As DateTime
        Dim fechaHasta As DateTime

        If DateTime.TryParse(txtFechaDesde.Text, fechaDesde) AndAlso DateTime.TryParse(txtFechaHasta.Text, fechaHasta) Then
            parametros.Add(txtFechaDesde.Text)
            parametros.Add(txtFechaHasta.Text)
        Else
            parametros.Add(Nothing)
            parametros.Add(Nothing)
        End If

        If Not String.IsNullOrEmpty(cboPrioridades.Text) Then
            Dim prioridad = cboPrioridades.SelectedValue.ToString()
            parametros.Add(prioridad)
        Else
            parametros.Add(Nothing)
        End If

        If Not String.IsNullOrEmpty(cboCriticidades.Text) Then
            Dim criticidad = cboCriticidades.SelectedValue.ToString()
            parametros.Add(criticidad)
        Else
            parametros.Add(Nothing)
        End If

        If Not String.IsNullOrEmpty(cboProductos.Text) Then
            Dim producto = cboProductos.SelectedValue.ToString()
            parametros.Add(producto)
        Else
            parametros.Add(Nothing)
        End If

        If Not String.IsNullOrEmpty(cboEstados.Text) Then
            Dim idEstado = cboEstados.SelectedValue.ToString()
            parametros.Add(idEstado)
        Else
            parametros.Add(Nothing)
        End If

        If Not String.IsNullOrEmpty(cboAsignadoA.Text) Then
            Dim asignadoA = cboAsignadoA.SelectedValue.ToString()
            parametros.Add(asignadoA)
        Else
            parametros.Add(Nothing)
        End If

        dgvBugs.DataSource = oBugService.consultarBugsConFiltros(parametros)

        If dgvBugs.Rows.Count = 0 Then
            MessageBox.Show("No se encontraron coincidencias para el/los filtros ingresados", "Aviso", MessageBoxButtons.OK, MessageBoxIcon.Information)

            btn_asignar.Enabled = False
            btn_editar.Enabled = False
            btn_detalle_bug.Enabled = False
            btn_pasar_a_te.Enabled = False
            btn_cerrar.Enabled = False
        End If
    End Sub

    Private Sub btn_detalle_bug_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_detalle_bug.Click
        Dim id As Integer

        If row_selected.Index <> -1 Then
            id = Convert.ToInt32(row_selected.Cells("id_bug_col").Value)
            Dim frmDetalle As frmDetalleBug = New frmDetalleBug()
            frmDetalle.mostrarBug(oBugService.consultarBugsPorId(id))
            frmDetalle.ShowDialog()
        End If
    End Sub

    Private Sub dgvBugs_CellClick(ByVal sender As Object, ByVal e As DataGridViewCellEventArgs) Handles dgvBugs.CellClick
        row_selected = dgvBugs.CurrentRow
        btn_asignar.Enabled = True
        btn_editar.Enabled = True
        btn_detalle_bug.Enabled = True
        btn_cerrar.Enabled = True
    End Sub

    Private Sub btn_cerrar_Click(ByVal sender As Object, ByVal e As EventArgs)

    End Sub

    Private Sub btn_salir_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_salir.Click
        Me.Close()
    End Sub

    Private Sub btn_editar_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_editar.Click
        Dim id As Integer

        If row_selected.Index <> -1 Then
            id = Convert.ToInt32(row_selected.Cells("id_bug_col").Value)
            Dim frmActualizar As frmActualizarBug = New frmActualizarBug()
            frmActualizar.mostrarBug(oBugService.consultarBugsPorId(id))
            frmActualizar.ShowDialog()
            btnConsultar_Click(sender, e)
        End If
    End Sub


    Private Sub btn_asignar_Click(sender As Object, e As EventArgs) Handles btn_asignar.Click
        Dim id As Integer

        If row_selected.Index <> -1 Then
            id = Convert.ToInt32(row_selected.Cells("id_bug_col").Value)
            Dim frmAsignar As frmAsignarBug = New frmAsignarBug()
            frmAsignar.mostrarBug(oBugService.consultarBugsPorId(id))
            frmAsignar.ShowDialog()
        End If
    End Sub
End Class

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmBugs.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer.Services;
using TPS_BugTracker.GUILayer.Helper;

namespace TPS_BugTracker.GUILayer.Bugs
{
    public partial class frmBugs : Form
    {
        private BugService oBugService = new BugService();

        private CriticidadService criticidadService = new CriticidadService();
        private PrioridadService prioridadService = new PrioridadService();
        private ProductoService productoService = new ProductoService();
        private UsuarioService usuarioService = new UsuarioService();
        private EstadoService estadoService = new EstadoService();

        internal DataGridViewRow row_selected;

        public frmBugs()
        {
            InitializeComponent();
            dgvBugs.AutoGenerateColumns = false;

        }

    
        private void frmBugs_Load(object sender, EventArgs e)
        {

            //LLenar combos y limpiar grid
            GUIHelper.getHelper().llenarCombo(cboEstados, estadoService.consultarEstados(), "nombre", "id_estado");

            GUIHelper.getHelper().llenarCombo(cboPrioridades, prioridadService.consultarPrioridades(), "nombre", "id_prioridad");

            GUIHelper.getHelper().llenarCombo(cboCriticidades, criticidadService.consultarCriticidades(), "nombre", "id_criticidad");

            GUIHelper.getHelper().llenarCombo(cboAsignadoA, usuarioService.consultarUsuarios(), "nombre", "id_usuario");

            GUIHelper.getHelper().llenarCombo(cboProductos, productoService.consultarProductos(), "nombre", "id_producto");

        }

        private void btnConsultar_Click(object sender, EventArgs e)
        {
            List<object> parametros = new List<object>();

            DateTime fechaDesde;
            DateTime fechaHasta;
            if(DateTime.TryParse(txtFechaDesde.Text, out fechaDesde) && 
                DateTime.TryParse(txtFechaHasta.Text, out fechaHasta)){
                parametros.Add(txtFechaDesde.Text);
                parametros.Add(txtFechaHasta.Text);
                }
            else
            {
                parametros.Add(null);
                parametros.Add(null);
            }

            if (!string.IsNullOrEmpty(cboPrioridades.Text))
            {
                var prioridad = cboPrioridades.SelectedValue.ToString();
                parametros.Add(prioridad);
            }
            else
            {
                parametros.Add(null);
            }

            if (!string.IsNullOrEmpty(cboCriticidades.Text))
            {
                var criticidad = cboCriticidades.SelectedValue.ToString();
                parametros.Add(criticidad);
            }
            else
            {
                parametros.Add(null);
            }

            if (!string.IsNullOrEmpty(cboProductos.Text))
            {
                var producto = cboProductos.SelectedValue.ToString();
                parametros.Add(producto);
            }
            else
            {
                parametros.Add(null);
            }


            if (!string.IsNullOrEmpty(cboEstados.Text))
            {
                var idEstado = cboEstados.SelectedValue.ToString();
                parametros.Add(idEstado);
            }
            else
            {
                parametros.Add(null);
            }

            if (!string.IsNullOrEmpty(cboAsignadoA.Text))
            {
                var asignadoA = cboAsignadoA.SelectedValue.ToString();
                parametros.Add(asignadoA);
            }
            else
            {
                parametros.Add(null);
            }
            
            dgvBugs.DataSource = oBugService.consultarBugsConFiltros( parametros);
            if(dgvBugs.Rows.Count == 0) 
            {
                MessageBox.Show("No se encontraron coincidencias para el/los filtros ingresados", "Aviso", MessageBoxButtons.OK, MessageBoxIcon.Information);

                btn_asignar.Enabled = false;
                btn_editar.Enabled = false;
                btn_detalle_bug.Enabled = false;
                btn_pasar_a_te.Enabled = false;
                btn_cerrar.Enabled = false;
            }

        }

        private void btn_detalle_bug_Click(object sender, EventArgs e)
        {
            int id;
            if (row_selected.Index != -1)
            {
                id = Convert.ToInt32(row_selected.Cells["id_bug_col"].Value);
                frmDetalleBug frmDetalle = new frmDetalleBug();
                frmDetalle.mostrarBug(oBugService.consultarBugsPorId(id));
                frmDetalle.ShowDialog();
            }
        }

        private void dgvBugs_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            row_selected = dgvBugs.CurrentRow;
            btn_asignar.Enabled = true;
            btn_editar.Enabled = true;
            btn_detalle_bug.Enabled = true;
            btn_cerrar.Enabled = true;
        }
        
        private void btn_salir_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btn_editar_Click(object sender, EventArgs e)
        {
            int id;
            if (row_selected.Index != -1)
            {
                id = Convert.ToInt32(row_selected.Cells["id_bug_col"].Value);
                frmActualizarBug frmActualizar = new frmActualizarBug();
                frmActualizar.mostrarBug(oBugService.consultarBugsPorId(id));
                frmActualizar.ShowDialog();

                btnConsultar_Click(sender, e);
            }
        }

        private void btn_asignar_Click(object sender, EventArgs e)
        {
            int id;
            if (row_selected.Index != -1)
            {
                id = Convert.ToInt32(row_selected.Cells["id_bug_col"].Value);
                frmAsignarBug frmAsignar = new frmAsignarBug();
                frmAsignar.mostrarBug(oBugService.consultarBugsPorId(id));
                frmAsignar.ShowDialog();
            }
        }

        private void btn_nuevo_Click(object sender, EventArgs e)
        {
            frmNuevoBug formulario = new frmNuevoBug();
            formulario.ShowDialog();
        }
    }
}
        